### PR TITLE
[civ2][flakiness/11] port python/ray/serve tests over to civ02

### PR DESCRIPTION
**Pull request by can-anyscale**
Port python/ray/serve tests over to civ02. There are other flavors or these tests; as well as debug and asan tests in core, etc. I'll port them over later.

Test:
- CI

### CODE REVIEW

**Comment by can-anyscale** [comment]
Screenshot 2023-08-03 at 8 01 26 AM

**Review by aslonnie — COMMENTED** [review]
this looks ok, but I would like to wait for `/10` to get merged first.

File: ci/ray_ci/serve.tests.env.sh
Comment (aslonnie) [inline ci/ray_ci/serve.tests.env.sh:10]: you probably do not need to unset these. on civ1 pipelines, it is still using dind technically speaking... and because you are running with your concurrent docker wrapper thing, these env vars are not set here anyways.

File: .buildkite/pipeline.build.yml
Comment (aslonnie) [inline .buildkite/pipeline.build.yml:42]: maybe add a comment on where to find the other tests.

**Review by aslonnie — COMMENTED** [review]
add a reviewer from serve team? @edoakes maybe?

File: ci/ray_ci/serve.tests.yml
Comment (edoakes) [inline ci/ray_ci/serve.tests.yml:1]: For my own education, what will happen to tests marked flaky going forward?

**Review by edoakes — APPROVED** [review]
We should hopefully be able to remove the two feature flagged-versions of the build soon. In the meantime I think it's still worth also porting them so all the tests run & report results in the same way.

File: ci/ray_ci/serve.tests.yml
Comment (can-anyscale) [reply ci/ray_ci/serve.tests.yml:1]: Technically, they will just become non-blocking. This is a way for team/test owner to unblock themselves if their tests suddenly become failing and blocking their PR without their faults. I want to see if test owners can manage their tests effectively this way, as well as not abusing this feature too much. If not, or if they want to make thing easier for them to manage, we can enforce through policy and automation down the line.

Comment (edoakes) [reply ci/ray_ci/serve.tests.yml:1]: Sounds good! I'll be happy to keep this list empty for the serve tests :D

Comment (can-anyscale) [reply ci/ray_ci/serve.tests.yml:1]: fantastic!

**Pull request merged**
